File: leetcode/easy/1656. Design an Ordered Stream/solution.cc
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class OrderedStream {
public:
    vector<string> stream;
    int pos = 0;
    OrderedStream(int n) {
        stream.resize(n);
    }
    
    vector<string> insert(int idKey, string value) {
        vector<string> result;
        stream[idKey - 1] = value;
        while (pos < stream.size() && stream[pos] != "") {
            result.push_back(stream[pos]);
            pos += 1;
        }
        return result;
    }
};

int main() {
    OrderedStream* OS = new OrderedStream(5);
    OS->insert(3, "ccc");
    OS->insert(1, "aaa");
    OS->insert(2, "bbb");
    OS->insert(5, "eee");
    OS->insert(4, "ddd");
    return 0;
}

/**
 * Your OrderedStream object will be instantiated and called as such:
 * OrderedStream* obj = new OrderedStream(n);
 * vector<string> param_1 = obj->insert(idKey,value);
 */

File: leetcode/easy/1656. Design an Ordered Stream/solution.js
/**
 * @param {number} n
 */
const OrderedStream = function(n) {
    this.stream = new Array(n);
    this.pos = 0;
};

/** 
 * @param {number} idKey 
 * @param {string} value
 * @return {string[]}
 */
OrderedStream.prototype.insert = function(idKey, value) {
    this.stream[idKey - 1] = value;
    let strings =  [];
    while (this.stream[this.pos]) {
        strings.push(this.stream[this.pos]);
        this.pos += 1;
    }
    return strings;
};

/** 
 * Your OrderedStream object will be instantiated and called as such:
 * var obj = new OrderedStream(n)
 * var param_1 = obj.insert(idKey,value)
 */

const stream = new OrderedStream(5);
console.log(stream.insert(3, 'ccc'));
console.log(stream.insert(1, 'aaa'));
console.log(stream.insert(2, 'bbb'));
console.log(stream.insert(5, 'eee'));
console.log(stream.insert(4, 'ddd'));